unleash-yggdrasil/src/lib.rs

```diff
@@ -246,24 +248,14 @@ pub struct ResolvedToggle {
 }
 
 impl EngineState {
-    pub fn take_delta(&mut self, delta: &ClientFeaturesDelta) -> Option<Vec<EvalWarning>> {
-        let mut current_state = self.compiled_state.take().unwrap_or_default();
-        let segment_map = build_segment_map(&delta.segments);
-        let mut warnings: Vec<EvalWarning> = vec![];
-        for removed in delta.removed.clone() {
-            current_state.remove(&removed);
-        }
-        for update in delta.updated.clone() {
-            let updated_state = compile(&update, &segment_map, &mut warnings);
-            current_state.insert(update.name.clone(), updated_state);
-        }
-        self.compiled_state = Some(current_state);
-        if warnings.is_empty() {
-            None
-        } else {
-            Some(warnings)
-        }
+    pub fn take_delta(&mut self, delta: &ClientFeaturesDelta) -> () {
+        let current_state = self.previous_state.clone();
+        let new_state = current_state.modify_and_copy(delta);
```

Comment:
Since we're cloning on the previous line, can this just be modify in place now?

Reply:
Looking at that more closely, it seems like I need to do a clone on `self.previous_state` unless there are other smart ways to do it I'm not aware of.

Reply:
Something like this:

```
       let mut new_state = self.previous_state.clone();
        new_state.modify_in_place(delta);
```